cluster/kubeconfig-creator.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: kubeconfig-creator
+spec:
+  params:
+    - name: name
+      description: name of the cluster
+      type: string
+    - name: type
+      description: type of the resource
+      type: string
+      default: cluster
+    - name: url
+      description: address of the cluster
+      type: string
+    - name: username
+      description: username for basic authentication to the cluster
+      type: string
+    - name: password
+      description: password for basic authentication to the cluster
+      type: string
+      default: ""
+    - name: cadata
+      description: contains PEM-encoded certificate authority certificates
+      type: string
+      default: ""
+    - name: clientKeyData
+      description: contains PEM-encoded data from a client key file for TLS
+      type: string
+      default: ""
+    - name: clientCertificateData
+      description: contains PEM-encoded data from a client cert file for TLS
+      type: string
+      default: ""
+    - name: namespace
+      description: default namespace to use on unspecified requests
+      type: string
+      default: ""
+    - name: token
+      description: bearer token for authentication to the cluster
+      type: string
+      default: ""
+    - name: insecure
+      description: to indicate server should be accessed without verifying the TLS certificate
+      type: string
+      default: "true"
+  workspaces:
+    - name: kubeconfigFile
+  steps:
+    - name: write
+      image: quay.io/diagrawa/kubeconfigwriter:v3
```

Comment:
If this is intended to act as a drop-in replacement for the Cluster Resource then I think we should continue to use Tekton's kubeconfigwriter image (`gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/kubeconfigwriter`).

What differences are there in this third-party image from the pipeline version?  Is there any reason we can't include those differences in pipeline's `cmd/kubeconfigwriter` package?

Reply:
`gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/kubeconfigwriter` image doesn't have `clientKeyData` and `clientCertificateData` in the  Resource struct. So need to build new image and change the `kubeconfigwriter` code.

Reply:
I have modified the `cmd/kubeconfigwriter` and will add to the required changes to this task, once we get next release for the pipeline. 
Meanwhile, can you please review this PR?